experiment/coverage_utils.py

```diff
@@ -292,3 +406,170 @@ def extract_covered_regions_from_summary_json(summary_json_file):
     except Exception:  # pylint: disable=broad-except
         logger.error('Coverage summary json file defective or missing.')
     return covered_regions
+
+
+def prepare_name_dataframes(df_container):
+    """Populating DataFrames with experiment specific
+    benchmark names, file names, and function names."""
+
+    # Collect segment and function CSV for all benchmark-fuzzer combinations.
+    try:
+        # Separate file names into a different CSV file and allocate ID's
+        # for each unique file name.
+        df_container.filename_df['file_name'] = \
+            df_container.segment_df['file_name'].unique()
+        df_container.filename_df.reset_index()
+        df_container.filename_df['file_id'] = \
+            df_container.filename_df.index + 1
+
+        # Separate function names into a different CSV and allocate ID's
+        # for each unique function name.
+        df_container.function_name_df['function_name'] = \
+            df_container.function_df['function_name'].unique()
+        df_container.function_name_df.reset_index()
+        df_container.function_name_df['function_id'] = \
+            df_container.function_name_df.index + 1
+
+        # Separate benchmark names into a different CSV and allocate ID's
+        # for each unique benchmark name.
+        df_container.benchmark_name_df['benchmark'] = \
+            df_container.segment_df['benchmark'].unique()
+        df_container.benchmark_name_df.reset_index()
+        df_container.benchmark_name_df['benchmark_id'] = \
+            df_container.benchmark_name_df.index + 1
+
+        # Separate fuzzer names into a different CSV and allocate ID's
+        # for each unique fuzzer name.
+        df_container.fuzzer_name_df['fuzzer'] = \
+            df_container.segment_df['fuzzer'].unique()
+        df_container.fuzzer_name_df.reset_index()
+        df_container.fuzzer_name_df['fuzzer_id'] = \
+            df_container.fuzzer_name_df.index + 1
+
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I don't like this at all. What are the errors that can occur here and why do they occur?
broad except should probably only be used in rare cases where things can be retried but not catching will stop the experiment.